File: Android/ProjectoESeminario/app/src/main/java/andre/pt/projectoeseminario/REST/IAPI.java
package andre.pt.projectoeseminario.REST;


import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.PUT;

public interface IAPI {

    String main = "http://projecto-testes.herokuapp.com/api/";
    String push = "push";
    String pull = "pull";
    String account = "account";

    @GET(account)
    boolean createAccount();

    @PUT(account)
    boolean authenticate(@Body("username") String username);


}

File: Windows/Projecto/Projecto/Service/Interfaces/IAPI.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net.Http;
using System.Text;
using System.Threading.Tasks;


namespace Projecto.Service
{

    public interface IAPI
    {
        void Push(long account, String data);
        Task<HttpResponseMessage> Pull(long account);
        Task<HttpResponseMessage> Authenticate(String username, String password);
        Task<HttpResponseMessage> CreateAccount(String username, String password);

    }
}
